File: 415-add-strings/add-strings.cpp
class Solution {
public:
    int charToInt(char s){
        return s - '0';
    }

    char intToChar(int a){
        return a + '0';
    }

    string addStrings(string num1, string num2) {
        if (num1.length() > num2.length())
            swap(num1, num2);

        reverse(num1.begin(), num1.end());
        reverse(num2.begin(), num2.end());
        string result = "";
        int carry = 0;

        for (int i = 0; i < num1.length(); i++) {
            int a = charToInt(num1[i]);
            int b = charToInt(num2[i]);
            int sum = a + b + carry;
            int ans = sum % 10;
            carry = sum / 10;
            result += intToChar(ans);
        }

        for (int i = num1.length(); i < num2.length(); i++) {
            int a = charToInt(num2[i]);
            int sum = a + carry;
            int ans = sum % 10;
            carry = sum / 10;
            result += intToChar(ans);
        }
        if (carry)
            result += "1";
        reverse(result.begin(), result.end());

        return result;
    }
};

File: 415-add-strings/add-strings.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        sys.set_int_max_str_digits(10000)
        return str(int(num1) + int(num2))

        """
        result = ""
        l1, l2, carry = len(num1) - 1, len(num2) - 1, 0

        while l1 >= 0 or l2 >= 0 or carry > 0:
            n1 = int(num1[l1]) if l1 >= 0 else 0
            n2 = int(num2[l2]) if l2 >= 0 else 0
            add = n1 + n2 + carry
            carry = add // 10
            result = str(add % 10) + result
            l1 -= 1
            l2 -= 1

        return result
        """
